src/features/SendAssets/components/SendAssetsForm.tsx

```diff
@@ -361,432 +363,383 @@ const SendAssetsForm = ({ txConfig, onComplete }: IStepComponentProps) => {
+        setFieldValue('gasEstimates', data);
+        setFieldValue('gasPriceSlider', data.fast);
+      });
+    }
+  });
+
+  useEffect(() => {
+    if (ptxState.protectTxShow) {
+      if (goToInitialStepOrFetchReport && ptxState.receiverAddress !== values.address.value) {
+        const { address, network } = values;
+        goToInitialStepOrFetchReport(address.value, network);
+      }
+    }
+  }, [values.address.value]);
+
+  const toggleAdvancedOptions = () => {
+    setFieldValue('advancedTransaction', !values.advancedTransaction);
+  };
+  const toggleIsAutoGasSet = () => {
+    // save value because setFieldValue method is async and values are not yet updated	            // save value because setFieldValue method is async and values are not yet updated
+    const isEnablingAutoGas = !values.isAutoGasSet;
+    setFieldValue('isAutoGasSet', !values.isAutoGasSet);
+    if (isEnablingAutoGas) {
+      handleGasEstimate(true);
+    }
+  };
+  const handleGasEstimate = async (forceEstimate: boolean = false) => {
+    if (
+      values &&
+      values.network &&
+      values.asset &&
+      values.address &&
+      isValidETHAddress(values.address.value) &&
+      values.account &&
+      isValidPositiveNumber(values.amount) &&
+      (values.isAutoGasSet || forceEstimate)
+    ) {
+      setIsEstimatingGasLimit(true);
+      const finalTx = processFormForEstimateGas(values);
+      const gas = await getGasEstimate(values.network, finalTx);
+      setFieldValue('gasLimitField', hexToNumber(gas).toString());
+      setFieldTouched('amount');
+      setIsEstimatingGasLimit(false);
+    } else {
+      return;
+    }
+  };
+
+  const handleFieldReset = () => {
+    const resetFields: (keyof IFormikFields)[] = [
+      'account',
+      'amount',
+      'txDataField',
+      'advancedTransaction'
+    ];
+    resetFields.forEach((field) => setFieldValue(field, initialValues[field]));
+  };
 
-          useEffect(() => {
-            handleNonceEstimate(values.account);
-          }, [values.account]);
+  const setAmountFieldToAssetMax = () => {
+    const account = getAccount(values.account);
+    if (values.asset && account && baseAsset) {
+      const accountBalance = getAccountBalance(account, values.asset).toString();
+      const isERC20 = isERC20Tx(values.asset);
+      const balance = fromTokenBase(new BN(accountBalance), values.asset.decimal);
+      const gasPrice = values.advancedTransaction ? values.gasPriceField : values.gasPriceSlider;
+      const amount = isERC20 // subtract gas cost from balance when sending a base asset
+        ? balance
+        : baseToConvertedUnit(
+            new BN(convertedToBaseUnit(balance.toString(), DEFAULT_ASSET_DECIMAL))
+              .sub(gasStringsToMaxGasBN(gasPrice, values.gasLimitField))
+              .toString(),
+            DEFAULT_ASSET_DECIMAL
+          );
+      setFieldValue('amount', amount);
+      handleGasEstimate();
```

Comment:
Move all gasEstimates inside an effect.

Reply:
Not sure we can move all of them inside an effect. I think we specifically wait for `blur` on the amount f.x. Since we don't want to spam update it.

Reply:
Otherwise we would update gas for every character entered in the amount field.